File: askdog/askdog-service/src/main/java/com/askdog/service/EventService.java
package com.askdog.service;

import com.askdog.service.bo.common.PagedData;
import com.askdog.service.bo.event.AmendedEvent;
import com.askdog.service.bo.event.EventDetail;
import com.askdog.service.bo.event.PureEvent;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import static org.springframework.web.bind.annotation.RequestMethod.*;

@FeignClient("service")
@RequestMapping("/service/event")
public interface EventService {

    @RequestMapping(method = POST)
    void createEvent(@RequestParam("userId") Long userId, @RequestBody PureEvent pureEvent);

    @RequestMapping(method = PUT)
    void updateEvent(@RequestParam("userId") Long userId, @RequestBody AmendedEvent amendedEvent);

    @RequestMapping(path = "/{eventId}", method = GET)
    EventDetail getEvent(@PathVariable("eventId") long eventId);

    @RequestMapping(path = "/{eventId}", method = DELETE)
    void deleteEvent(@RequestParam("userId") Long userId, @PathVariable("eventId") long eventId);

    @RequestMapping(path = "/search", method = POST)
    PagedData<EventDetail> getEvents(@RequestParam("userId") Long userId, @RequestParam("storeId") long storeId, @RequestBody Pageable pageable);

}

File: askdog-web-coupon-store/src/js/service/EventService.js
define(['base/BaseService'], function (BaseService) {

    var EventService = BaseService.extend({

        init: function (http, scope) {
            this._super(http, scope);
        },

        createEvent: function(event) {
            return this.post("/api/events", event);
        },

        modifyEvent: function(event) {
            return this.put("/api/events", event);
        },

        deleteEvent: function(eventId) {
            return this.delete("/api/events/{0}".format(eventId));
        },

        search: function(storeId, page, size) {
            return this.get("/api/events/search?storeId={storeId}&page={page}&size={size}".format({
                storeId: storeId,
                page: page,
                size: size
            }));
        }
    });

    angular.module('service.EventService', [])
        .provider('EventService', function () {
            this.$get = ['$http', '$rootScope', function (http, scope) {
                return new EventService(http, scope);
            }];
        });

});